apis/bigquery/v1beta1/dataset_reference.go

```diff
@@ -71,10 +72,42 @@ func (r *DatasetRef) NormalizedExternal(ctx context.Context, reader client.Reade
 		return "", fmt.Errorf("reading referenced %s %s: %w", BigQueryDatasetGVK, key, err)
 	}
 	// Get external from status.externalRef. This is the most trustworthy place.
-	actualExternalRef, _, err := unstructured.NestedString(u.Object, "status", "externalRef")
+	actualExternalRef, found, err := unstructured.NestedString(u.Object, "status", "externalRef")
 	if err != nil {
 		return "", fmt.Errorf("reading status.externalRef: %w", err)
 	}
+	if !found {
```

Comment:
What happens if the KRM YAML is applied to the cluster, but the KCC controller has not yet reconciled it? In this scenario, the `externalRef` would be missing, and the BigQueryDataset GCP resource is not yet created.

Reply:
Yes and that's okay because it will be found eventually! (Say a few reconciles later after)

Reply:
This actually represents a significant behavior change on the KCC side. In the scenario I described, there is no GCP resource yet, but we are sending an HTTP request that references it. According to KCC conventions, if a dependent resource is not ready (e.g., still in the process of being created), the KCC controller waits until it is ready. The new behavior, however, causes the KCC controller to continuously send HTTP requests to the GCP API until the dependent resource is created in GCP.

Reply:
I've parked the findings from an offline discussion here: https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/3619

I think the current PR can be merged as is (in particular with the fix for `IsResourceReady`).